### DISCUSSION TITLE
关于ASSISTments2017这个数据集

### DISCUSSION THREAD

**Discussion by wulongxiaomeem** [pykt-team/pykt-toolkit#71] [Q&A]
请问试题和skill(知识点)的对应关系是什么呢？我自己的观察是觉得一个题目可以对应多个skill，但一个用户在解决这个问题时只会用到一个skill，并且不同的用户解决同个问题时用的skill不同，不知道有没有错误。另外，在这种情况下该如何设置Q矩阵呢？

**Comment by Liu-lqq** [comment]
对应关系在各个数据集官方提供的最原始的数据表中。
我们在训练时，对于一个题目存在多个知识点的情况，会将一个question-answer的交互复制扩充为多个skill-answer的交互，在进行预测时，分别预测各个知识点的作答准确率，之后进行fusion，细节可见pykt/preprocess/split_datasets.py和pykt/models/evaluate_model.py。也可参考我们的论文：https://arxiv.org/pdf/2206.11460.pdf

**Reply by wulongxiaomeem** [reply]
感谢回复！！！
